### PR TITLE
import export SharedPreferences

### PR DESCRIPTION
**Pull request by albertsal**
i add function for import and export SharedPreferences setting

when export or import db execute export or import SharedPreferences

for blacklist i preload HashSet

### CODE REVIEW

**Comment by albertsal** [comment]
I @jpbarraca valute this pr?

**Comment by jpbarraca** [comment]
@albertsal I currently only vouching PRs for HPlus devices

**Comment by albertsal** [comment]
ok ;)

**Comment by cpfeiffer** [comment]
I'll have a look in the next days. Was busy with other things, sorry.

**Comment by albertsal** [comment]
ok do not worry @cpfeiffer

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/ImportExportSharedPreferences.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import static nodomain.freeyourgadget.gadgetbridge.GBApplication.blacklist;
+
+public class ImportExportSharedPreferences {
+
+    private static final String BOOLEAN = Boolean.class.getSimpleName();
+    private static final String FLOAT = Float.class.getSimpleName();
+    private static final String INTEGER = Integer.class.getSimpleName();
+    private static final String LONG = Long.class.getSimpleName();
+    private static final String STRING = String.class.getSimpleName();
+    private static final String HASTSET = HashSet.class.getSimpleName();
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/ImportExportSharedPreferences.java]: Typo?

Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/ImportExportSharedPreferences.java]: Are these the only supported classes supported by SharedPreferences?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java [hunk 2]
```diff
@@ -1,27 +1,12 @@
-/*  Copyright (C) 2016-2017 Andreas Shimokawa, Carsten Pfeiffer, Daniele
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java]: Can you add this back, please? :)

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java [hunk 3]
```diff
@@ -110,8 +101,33 @@ private String getExternalPath() {
...
 
+    private void exportShared() {
+        // BEGIN EXAMPLE
+        File myPath = null;
+        try {
+            myPath = FileUtils.getExternalFilesDir();
+            File myFile = new File(myPath, "Export_preference");
+            shared_file.exportToFile(sharedPrefs,myFile,null);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java]: These exceptions should probably be promoted to user notifications (e.g. toasts at least).

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/ImportExportSharedPreferences.java (hunk 1 above)
Comment (albertsal) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/ImportExportSharedPreferences.java]: no, I have not found in other types gb

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java (hunk 2 above)
Comment (albertsal) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DbManagementActivity.java]: ok

**Comment by cpfeiffer** [comment]
Thanks a lot, Alberto!

**Comment by albertsal** [comment]
;)

**Pull request merged**